inc/com/centreon/engine/downtimes/downtime_manager.hh

```diff
@@ -0,0 +1,126 @@
+                                      bool               fixed,
+                                      unsigned long      triggered_by,
+                                      unsigned long      duration,
+                                      unsigned long*     downtime_id);
+  std::shared_ptr<downtime>         add_service_downtime(
+                                      std::string const& host_name,
+                                      std::string const& svc_description,
+                                      time_t             entry_time,
+                                      char const*        author,
+                                      char const*        comment_data,
+                                      time_t             start_time,
+                                      time_t             end_time,
+                                      bool               fixed,
+                                      unsigned long      triggered_by,
+                                      unsigned long      duration,
+                                      unsigned long      downtime_id);
+  std::shared_ptr<downtime>         add_new_service_downtime(
+                                      std::string const& host_name,
+                                      std::string const& service_description,
+                                      time_t entry_time,
+                                      char const* author,
+                                      char const* comment_data,
+                                      time_t start_time,
+                                      time_t end_time,
+                                      bool fixed,
+                                      unsigned long triggered_by,
+                                      unsigned long duration,
+                                      unsigned long* downtime_id);
+  int                               schedule_downtime(
+                                      int                type,
+                                      std::string const& host_name,
+                                      std::string const& service_description,
+                                      time_t             entry_time,
+                                      char const*        author,
+                                      char const*        comment_data,
+                                      time_t             start_time,
+                                      time_t             end_time,
+                                      bool               fixed,
+                                      unsigned long      triggered_by,
+                                      unsigned long      duration,
+                                      unsigned long*     new_downtime_id);
+
+ private:
+                                    downtime_manager() = default;
+  std::multimap<time_t, std::shared_ptr<downtime>>
+                                    _scheduled_downtimes;
```

Comment:
this type look widely use. Perhaps you can typedef a downtime_multimap outside of CCE namepace. Like what is done in contact or contactgroup ?

Reply:
Yes, why not.